lib/src/iterable_matchers.dart

```diff
@@ -206,6 +216,31 @@ class _UnorderedMatches extends Matcher {
   Description describeMismatch(item, Description mismatchDescription,
           Map matchState, bool verbose) =>
       mismatchDescription.add(_test(item));
+
+  /// Returns [true] if the value at [valueIndex] can be paired with some
+  /// unmatched matcher and updates the state of [matched].
+  ///
+  /// If there is a conflic where multiple values may match the same matcher
+  /// recursively looks for a new place to match the old value. [seen] trackes
+  /// the matchers that have been used _during_ this search.
+  bool _findPairing(List<List<int>> edges, int valueIndex, List<int> matched,
+      [Set<int> seen]) {
+    seen ??= new Set<int>();
+    final possiblePairings = edges[valueIndex].where((m) => !seen.contains(m));
+    for (final matcherIndex in possiblePairings) {
+      seen.add(matcherIndex);
+      final previouslyMatched = matched[matcherIndex];
+      final canPlaceWithoutConflict = previouslyMatched == null;
```

Comment:
nit: I don't think this local var is necessary, or you could eliminate `previouslyMatched` and move the null check up

Reply:
Removed `canPlaceWithoutConflict`